src/App.tsx

```diff
@@ -1,9 +1,185 @@
-import './App.scss';
+import { useMemo, useState, useContext } from 'react';
+import { useNavigate } from 'react-router';
+import {
+  Navigate,
+  Route,
+  Routes,
+  useLocation,
+} from 'react-router-dom';
+import { Context } from './Context';
+import { Header } from './Header';
+import { Navigation } from './Navigation';
+import { Home } from './Home';
+import { Products } from './Products';
+import { ProductDetails } from './ProductDetails';
+import { Favourites } from './Favourites';
+import { Cart } from './Cart';
+import { Footer } from './Footer';
+import { ProductTypes } from './types/productTypes';
+import { SearchTypes } from './types/SearchTypes';
+import { LocaleStorageTypes } from './types/LocaleStorageTypes';
+import { useUpdateSearch } from './utils/hooks';
 
-const App = () => (
-  <div className="App">
-    <h1>React Phone Catalog</h1>
-  </div>
-);
+const App = () => {
+  const { setQuery } = useContext(Context);
+  const { pathname } = useLocation();
+  const history = useNavigate();
+  const { updateSearch, searchParams } = useUpdateSearch();
+  const [filterType, setFilterType] = useState('');
+  const page = searchParams.get(SearchTypes.page) || '';
+  const perPage = searchParams.get(SearchTypes.perPage) || '';
+  const sort = searchParams.get(SearchTypes.sort) || '';
+  const filterQuery = searchParams.get(SearchTypes.query) || '';
+  const activeProduct = JSON.parse(
+    localStorage.getItem(LocaleStorageTypes.product) as string,
+  ) || null;
+
+  useMemo(() => {
+    if (pathname === '/phones') {
+      setFilterType('phones');
+    } else if (pathname === '/tablets') {
+      setFilterType('tablets');
+    } else if (pathname === '/accessories') {
+      setFilterType('accessories');
+    } else if (pathname === '/favourites') {
+      setFilterType('favourites');
+    } else {
+      setFilterType('');
+    }
```

Comment:
move to useEffect 

try to read where we should use [useMemo](https://react.dev/reference/react/useMemo)